cpp/src/traversal/bfs_kernels.cuh

```diff
@@ -299,8 +299,9 @@ __global__ void main_bottomup_kernel(const IndexType *unvisited,
     // by different in in visited_bmap)
     IndexType visited_bmap_index[1];  // this is an array of size 1 because CUB
                                       // needs one
-    visited_bmap_index[0]      = -1;
-    IndexType unvisited_vertex = -1;
+
+    visited_bmap_index[0]      = ~IndexType(0);
+    IndexType unvisited_vertex = ~IndexType(0);
```

Comment:
See

https://github.com/rapidsai/cugraph/pull/838/files#diff-ba59ce02957abb53c35f1a1929fc0374R682

This is not merged yet (and you can just copy and paste this part and to graph.hpp), but I think this is a better way to handle this.

Reply:
Done. Added the invalid_idx type to graph.hpp, and invalid_vertex_id and invalid_edge_id derive from it. Btw, technically, there shouldn't be a reason to use -1 for int. std::numeric_limits<T>::max() should work for all types. The only place it will affect things is possibly at the API level where currently -1 is returned as invalid (for eg., for predecessors). In the future, it could be changed to be std::numeric_limits<T>::max() to make it uniform.